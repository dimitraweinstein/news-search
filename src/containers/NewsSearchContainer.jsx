import React, { Component } from 'react';

export default class NewsSearchContainer extends Component {
  state = {
    loading: true,
    articles: [],
  }
  render() {

    const { loading } = this.state;
    
    if(loading) return <p>Loading...</p>;
  }
}
