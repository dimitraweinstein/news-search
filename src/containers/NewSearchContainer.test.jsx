import React from 'react';
import { render, screen } from '@testing-library/react';
// import { rest } from 'msw';
// import { setupServer } from 'msw/node';
import userEvent from '@testing-library/user-event';
import NewsSearchContainer from './NewsSearchContainer';

// const server = setupServer(
//   rest.get(`https://newsapi.org/v2/everything?q=&apiKey=1035d26ed87d480dad7b5b19c00d27a9&qInTitle=${userQuery}`, (req, res, ctx) => {
//     return res(ctx.json());
//   })
// );

describe('NewsSearchContainer', () => {
  it('should display a list of news articles', async () => {
    render(<NewsSearchContainer />);

    screen.getByAltText('loading spinner');

    const input = await screen.findByLabelText('Search');
    userEvent.type(input, 'baseball');

    // eslint-disable-next-line max-len
    const submitButton = await screen.findByRole('button', { name: 'search-articles' });
    userEvent.click(submitButton);
    
    const ul = screen.findByRole('list', { name: 'articles' });
    expect(ul).toMatchSnapshot();    
    
  });
});
