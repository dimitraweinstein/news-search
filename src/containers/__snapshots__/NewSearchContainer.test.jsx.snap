// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsSearchContainer should display a list of news articles 1`] = `Promise {}`;
